### ISSUE TITLE
Change approach to auto-generating policy

### ISSUE THREAD

**Issue by funkybob** [awslabs/chalice#64]
Whilst getting started has been simple, now that I have some custom policies, it's become a pain.

I recommend the following:

On deploy, if .chalice/policy.json exists, use it.
If it doesn't, use .chalice/default-policy.json
If that doesn't, create it using "gen-policy" and/or the auto policy magicks.

This way, it's sill just as easy to get started, and not tedious to deploy once you've decided to write your own policy.json

[I fully admit I may be missing important details about filenames, etc... like why you safe policy.json instead of generating its content when needed if it doesn't exist]

**Comment by sdole** [comment]
@funkybob I had a similar issue... see #61 and a pull request I made for it #62.

My problem was that developers do not have access to creating IAM policies at will. So, I need chalice to use a pre-created IAM role instead of generating anything new. With the pull request, I am able to configure a role arn as so:
```
{
  "app_name": "myapp",
  "stage": "dev",
  "role_arn": "arn:aws:iam::<accountid>:role/<rolename>"
}
```

**Comment by jamesls** [comment]
If I understand the high level use case, what you'd like to do is manually create the exact policy you want to use, save it somewhere on disk (say `.chalice/some-file`), and not have to specify `--no-autogen-policy` every time you run `chalice deploy`.

Is that correct?

**Comment by funkybob** [comment]
That is correct.  It would be nice to not have to keep supplying that once I've clearly provided my own non-default policy.

**Comment by jamesls** [comment]
Building on the work from @sdole, I think the way I'd like to address this by adding another config option which is `autogen_policy`.  If this value is false, then it will use whatever is in `.chalice/policy.json`.  That way you have a few levels of control:

* default case - create the role if needed, autogen and set the policy for the role on each release.
* `autogen_policy` - Create the role as necessary, but do not autogen the policy.  Use whatever is in `.chalice/policy.json`.  If that file doesn't exist, and error is raised.
* `manage_iam_role` - Completely disable `chalice deploy` from creating/updating the role and any policies associated with the role.

**Comment by brylie** [comment]
It would be helpful to keep the policy document under revision control. That way, other parts of the policy could be configured, such as related permissions.
